### ISSUE TITLE
CSVReader ignores format passed

### ISSUE THREAD

**Issue by oschonrock** [vincentlaucsb/csv-parser#143]
for the simple case:
```C++
    csv::CSVFormat format;
    format.no_header();
    csv::CSVReader reader("filename.csv", format);
```

The `no_header()` call is ignored.

It turns out that the definition of that constructor (the one that takes a `string_view` for the filename)

https://github.com/vincentlaucsb/csv-parser/blob/621a9d94e4eab8acd31a8ac35577a68acdc1870f/include/internal/csv_reader.cpp#L153

is missing the constructor initialisation list to copy the passed `format` into the `_format` member (it only copies the passed `format` to `this->format` in the body of that constructor a few lines down if `format.guess_delim == true`)

I would suggest that csv_reader.cpp#L153 needs to be this instead - using the constructor initialisation list to always put a copy of the passed `format` into the `_format` member, and then overwriting it with a changed copy if `format.guess_delim == true`:
```C++
    CSV_INLINE CSVReader::CSVReader(csv::string_view filename, CSVFormat format) : _format(format) {                                                                      
```

This would be analogous to the constructor which takes a stream (defined inline, in the header file):

https://github.com/vincentlaucsb/csv-parser/blob/621a9d94e4eab8acd31a8ac35577a68acdc1870f/include/internal/csv_reader.hpp#L121

Making that change makes the test case at top work as expected for me.

**Comment by oschonrock** [comment]
looks a duplicate of this pull request:

https://github.com/vincentlaucsb/csv-parser/pull/137/commits/932ed2199e16840e99479f13c1c87fba722de7f0

**Comment by vincentlaucsb** [comment]
Fixed by #137

**Comment by oschonrock** [comment]
thanks
